File: Java/RecyclerView/RecyclerView-base/app/src/main/java/com/ls/project/recyclerview/MainActivity.java
package com.ls.project.recyclerview;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private RecyclerView mRcyclerView;
    private RecyclerView.Adapter mAdapter;
    private RecyclerView.LayoutManager mLayoutManager;
    private ArrayList<ListItem> items;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mRcyclerView = findViewById(R.id.recyclerView);
        mRcyclerView.setHasFixedSize(true);
        mLayoutManager = new LinearLayoutManager(this);
        mRcyclerView.setLayoutManager(mLayoutManager);

        items = new ArrayList<>();
        for (int i = 1; i < 10; i++) {
            items.add(new ListItem("구역 " + String.valueOf(i), String.valueOf(i) + " 층", String.valueOf(i * 10)));
        }
        mAdapter = new ListRecyclerViewAdapter(items);

        mRcyclerView.setAdapter(mAdapter);

    }
}

File: Kotlin/RecyclerView/RecycelerView-databinding/app/src/main/java/com/ls/project/areacard/MainActivity.kt
package com.ls.project.areacard

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import com.ls.project.areacard.adapter.ListRecyclerViewAdapter
import com.ls.project.areacard.bindings.BaseRecyclerViewAdapter
import com.ls.project.areacard.databinding.ActivityMainBinding
import com.ls.project.areacard.databinding.ItemListBinding
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private lateinit var items: ArrayList<ListItem>
    var adapter = BaseRecyclerViewAdapter<ListItem, ItemListBinding>(R.layout.item_list, BR.item)
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        items = arrayListOf<ListItem>()
        for (i in 0..10) {
            items.add(ListItem("구역 $i", "${i * 10}"))
        }


        binding.main = this
    }
}
